src/main/kotlin/com/quandoo/lib/reactivekafka/consumer/KafkaConsumer.kt

```diff
@@ -54,33 +55,50 @@ class KafkaConsumer {
 
     constructor(
         kafkaProperties: KafkaProperties,
-        kafkaListenerMetas: List<KafkaListenerMeta<*, *>>
+        originalKafkaListenerMetas: List<KafkaListenerMeta<*, *>>
     ) {
         checkKafkaProperties(kafkaProperties)
-        checkListeners(kafkaListenerMetas)
+        checkListeners(originalKafkaListenerMetas)
 
         this.kafkaProperties = kafkaProperties
-        this.kafkaListenerMetas = kafkaListenerMetas
+        this.kafkaListenerMetas = mergeConfiguration(kafkaProperties, originalKafkaListenerMetas)
         this.schedulers = kafkaListenerMetas.map { it to Schedulers.newParallel("kafka-consumer-${it.topics}", kafkaProperties.consumer!!.parallelism) }.toMap()
     }
 
+    private fun mergeConfiguration(kafkaProperties: KafkaProperties, listeners: List<KafkaListenerMeta<*, *>>): List<KafkaListenerMeta<*, *>> {
+        return listeners.map { kafkaListenerMeta ->
+            kafkaListenerMeta.copy(
```

Comment:
Can't we place a validation for those properties that should not be null after this copy process? I mean a few fields were called `Optional` but they are not in reality...